sonar-scm-git-plugin/src/main/java/org/sonarsource/scm/git/ChangedLinesComputer.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * SonarQube :: Plugins :: SCM :: Git
+ * Copyright (C) 2014-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.scm.git;
+
+import java.io.OutputStream;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+class ChangedLinesComputer {
+
+  private final Tracker tracker = new Tracker();
+
+  private final OutputStream receiver = new OutputStream() {
```

Comment:
This `OutputStream` looks strange. What is the expected encoding of the incoming bytes? Is it actually just the unicode int values?

Reply:
That's a very good question. My understanding is that Git dumps raw bytes. Note that we largely ignore the content, except for the block headers and the first characters per line, assuming `\n` terminating lines.

I guess we (uhm, I) should test this by adding files to a repo with funky encoding, and see if we are still able to distinguish the key parts of the output...